client-hc/src/main/java/com/graphhopper/api/GHMRequest.java

```diff
@@ -1,193 +1,163 @@
+        return pointHints;
     }
 
-    public GHRequest addFromPointHint(String pointHint) {
-        this.fromPointHints.add(pointHint);
-        return this;
-    }
-
-    public GHRequest setFromPointHints(List<String> pointHints) {
-        // create new array as we modify pointHints in compactPointHints
-        this.fromPointHints = new ArrayList<>(pointHints);
+    public GHMRequest setFromPointHints(List<String> fromPointHints) {
+        this.fromPointHints = fromPointHints;
         return this;
     }
 
     public List<String> getFromPointHints() {
         return fromPointHints;
     }
 
-    public GHRequest addToPointHint(String pointHint) {
-        this.toPointHints.add(pointHint);
-        return this;
-    }
-
-    public GHRequest setToPointHints(List<String> pointHints) {
-        // create new array as we modify pointHints in compactPointHints
-        this.toPointHints = new ArrayList<>(pointHints);
+    public GHMRequest setToPointHints(List<String> toPointHints) {
+        this.toPointHints = toPointHints;
         return this;
     }
 
     public List<String> getToPointHints() {
         return toPointHints;
     }
 
-    public GHMRequest addFromCurbside(String curbside) {
-        fromCurbsides.add(curbside);
+    public GHMRequest setCurbsides(List<String> curbsides) {
+        this.curbsides = curbsides;
         return this;
     }
 
-    public GHMRequest setFromCurbsides(List<String> curbsides) {
-        fromCurbsides = curbsides;
+    public List<String> getCurbsides() {
+        return curbsides;
+    }
+
+    public GHMRequest setFromCurbsides(List<String> fromCurbsides) {
+        this.fromCurbsides = fromCurbsides;
         return this;
     }
 
     public List<String> getFromCurbsides() {
         return fromCurbsides;
     }
 
-    public GHMRequest addToCurbside(String curbside) {
-        toCurbsides.add(curbside);
+    public GHMRequest setToCurbsides(List<String> toCurbsides) {
+        this.toCurbsides = toCurbsides;
         return this;
     }
 
-    public GHMRequest setToCurbsides(List<String> curbsides) {
-        toCurbsides = curbsides;
+    public List<String> getToCurbsides() {
+        return toCurbsides;
+    }
+
+    public GHMRequest setSnapPreventions(List<String> snapPreventions) {
+        this.snapPreventions = snapPreventions;
         return this;
     }
 
-    public List<String> getToCurbsides() {
-        return toCurbsides;
+    public List<String> getSnapPreventions() {
+        return snapPreventions;
     }
 
-    @Override
-    public GHRequest setCurbsides(List<String> curbsides) {
-        fromCurbsides = curbsides;
-        toCurbsides = curbsides;
+    // a good trick to serialize unknown properties into the HintsMap
+    @JsonAnySetter
+    public GHMRequest putHint(String fieldName, Object value) {
+        if (profile != null && "profile".equals(fieldName))
+            throw new IllegalArgumentException("use 'profile' only explicitly");
```

Comment:
Makes kind of sense, I guess but unfortunately this can easily be circumvented by using `getHints().putObject("profile", "car")`. So maybe rather check if "profile" is contained in the hints in `createPostRequest` later?

Reply:
And if we do this maybe the same for fail_fast as it is also proper field here and should be consistent with the hints.
... and maybe even a warning if 'vehicle' is still used? No, probably this is too much.